src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -187,8 +206,11 @@ impl Binder {
                                         as_of,
                                         true,
                                     );
-                                } else if let Some(table_catalog) = schema
-                                    .get_created_table_or_any_internal_table_by_name(table_name)
+                                } else if let Some(table_catalog) =
+                                    schema.get_table_by_name(table_name)
+                                    && (bind_creating_relations
+                                        || table_catalog.is_internal_table()
+                                        || table_catalog.is_created())
```

Comment:
Preserves the original semantics + accepts any table if `bind_creating_relations` is set.